### PR TITLE
fix(developer): remove redundant check in LdmlKeyboardCompiler.validate() 🙀 

### PR DESCRIPTION
**Pull request by srl295**
- simplify LdmlKeyboardCompiler.validate() by removing a redundant check

@keymanapp-test-bot skip

Suggested by https://github.com/keymanapp/keyman/pull/11810#discussion_r1650176743

### CODE REVIEW

File: developer/src/kmc-ldml/src/compiler/compiler.ts [hunk 1]
```diff
@@ -291,10 +291,14 @@ export class LdmlKeyboardCompiler implements KeymanCompiler {
...
 
   /**
-   * Runs any linter steps
+   * Runs any linter steps, adding hints to the callbacks as needed
    * @internal
+   * @returns true unless there was a linter failure.
    */
   private async lint(source: LDMLKeyboardXMLSourceFile, kmx: KMXPlus.KMXPlusFile) : Promise<boolean> {
+    if (!kmx || !source) {
+      return false;
```
Comment (jahorton) [inline developer/src/kmc-ldml/src/compiler/compiler.ts:300]: So... if there's no source or no kmx, that's a linter failure?

**Comment by mcdurdin** [comment]
> Fixes: #11054

I don't understand how this PR fixes that?

Could you update the title to clarify what is being done here?

File: developer/src/kmc-ldml/src/compiler/compiler.ts (hunk 1 above)
Comment (rc-swag) [reply developer/src/kmc-ldml/src/compiler/compiler.ts:300]: Well, I see that it isn't a linter failure just a broken pre-condition so bail returning false.

**Review by rc-swag — APPROVED** [review]
lgtm

**Comment by srl295** [comment]
> > Fixes: #11054
> 
> I don't understand how this PR fixes that?
> 
> Could you update the title to clarify what is being done here?

Updated.

I removed the "Fixes" - the feature work was in https://github.com/keymanapp/keyman/pull/11810 this is just an improvement to the fix.

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.92-alpha](https://keyman.com/downloads/releases/alpha/18.0.92)

**Pull request merged**